Source/SpatialGDK/Private/SpatialActorChannel.cpp

```diff
@@ -364,42 +365,44 @@ bool USpatialActorChannel::ReplicateActor()
 	{		
 		if (RepFlags.bNetInitial && bCreatingNewEntity)
 		{
-			// When a player is connected, a FUniqueNetIdRepl is created with the players worker ID. This eventually gets stored
-			// inside APlayerState::UniqueId when UWorld::SpawnPlayActor is called. If this actor channel is managing a pawn or a 
-			// player controller, get the player state.
-			FString PlayerWorkerId;
-			APlayerState* PlayerState = Cast<APlayerState>(Actor);
-			if (!PlayerState)
+			if (!Actor->IsFullNameStableForNetworking() || SpatialNetDriver->GetSpatialInterop()->CanSpawnReplicatedStablyNamedActors())
 			{
-				APawn* Pawn = Cast<APawn>(Actor);
-				if (Pawn)
+				// When a player is connected, a FUniqueNetIdRepl is created with the players worker ID. This eventually gets stored
+				// inside APlayerState::UniqueId when UWorld::SpawnPlayActor is called. If this actor channel is managing a pawn or a 
+				// player controller, get the player state.
+				FString PlayerWorkerId;
+				APlayerState* PlayerState = Cast<APlayerState>(Actor);
+				if (!PlayerState)
 				{
-					PlayerState = Pawn->PlayerState;
+					if (APawn* Pawn = Cast<APawn>(Actor))
+					{
+						PlayerState = Pawn->PlayerState;
+					}
 				}
-			}
-			if (!PlayerState)
-			{
-				if (PlayerController)
+				if (!PlayerState)
```

Comment:
I think this one is needed: PlayerState is first attempted as a cast from actor, then as a cast from the (Pawn)Actor->PlayerState and then as a (PlayerController)Actor->PlayerState, i.e. it can go into the first if, and not go in the second, because the PlayerState is now set.

Reply:
After discussion offline, this will be refactored.

Reply:
After further discussion offline, this will not be refactored.